plugins/obs-ffmpeg/texture-amf.cpp

```diff
@@ -2105,10 +2299,68 @@ try {
 	return false;
 }
 
+static inline void check_recommended_av1_defaults(amf_base *enc,
+						  obs_data_t *settings)
+{
+	const bool is10bit = enc->amf_format == AMF_SURFACE_P010;
+	const int64_t framerate = enc->fps_num / enc->fps_den;
+	if ((enc->cx * enc->cy > 1920 * 1088) || is10bit || (framerate > 60)) {
+		// Recommended base defaults
+		obs_data_set_default_int(settings, "bitrate", 2500);
+		obs_data_set_default_int(settings, "cqp", 20);
+		obs_data_set_default_string(settings, "rate_control", "VBR");
+		obs_data_set_default_string(settings, "preset", "balanced");
+		obs_data_set_default_string(settings, "profile", "main");
+		info("Original default settings were lowered according to resolution"
+		     " and framerate.");
+	} else {
+		// Recommended high perceptual quality defaults
+		// for up to 1080p60fps
+
+		bool pa_enabled = obs_data_get_bool(settings, "pre_analysis");
+		bool pa_set_by_user =
+			obs_data_has_user_value(settings, "pre_analysis");
+
+		if (!pa_set_by_user || (pa_set_by_user && pa_enabled)) {
+			pa_enabled = true;
+			uint64_t pa_lab_depth = 20;
+			AMF_MEMORY_TYPE pa_engine_type = AMF_MEMORY_OPENCL;
+			AMF_PA_TAQ_MODE_ENUM pa_taq_mode = AMF_PA_TAQ_MODE_2;
+			set_av1_property(enc, PRE_ANALYSIS_ENABLE, pa_enabled);
+			set_amf_property(enc, AMF_PA_LOOKAHEAD_BUFFER_DEPTH,
+					 pa_lab_depth);
+			set_amf_property(enc, AMF_PA_TAQ_MODE, pa_taq_mode);
+			set_amf_property(enc, AMF_PA_ENGINE_TYPE,
+					 AMF_MEMORY_OPENCL);
+			info("High perceptual quality defaults:\n"
+			     "\tAv1EnablePreAnalysis:   %s\n"
+			     "\tPALookAheadBufferDepth: %d\n"
+			     "\tPATemporalAQMode:       %d\n"
+			     "\tPAEngineType:           %s\n",
+			     pa_enabled ? "true" : "false", pa_lab_depth,
+			     pa_taq_mode,
+			     pa_engine_type == AMF_MEMORY_HOST     ? "Host"
+			     : pa_engine_type == AMF_MEMORY_DX11   ? "DX11"
+			     : pa_engine_type == AMF_MEMORY_OPENCL ? "OpenCL"
+			     : pa_engine_type == AMF_MEMORY_VULKAN ? "Vulkan"
+			     : pa_engine_type == AMF_MEMORY_UNKNOWN
+				     ? "Unknown (Auto)"
+				     : "Invalid");
```

Comment:
Consider a static inline function with a switch-case statement instead of chained ternaries.